packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { startCase } from 'lodash';
+import EquivalentKeyMap from 'equivalent-key-map';
+
+/**
+ * WordPress dependencies
+ */
+import { CheckboxControl, Modal, Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+const EntitiesSavedStatesCheckbox = ( {
+	id,
+	name,
+	changes: { rawRecord },
+	checked,
+	setCheckedById,
+} ) => (
+	<CheckboxControl
+		label={ `${ startCase( name ) }: "${ rawRecord.title ||
+			rawRecord.name ||
+			__( 'Untitled' ) }"` }
+		checked={ checked }
+		onChange={ ( nextChecked ) => setCheckedById( id, nextChecked ) }
+	/>
+);
+
+export default function EntitiesSavedStates( {
+	isOpen,
+	onRequestClose,
+	ignoredForSave = new EquivalentKeyMap(),
+} ) {
+	const entityRecordChangesByRecord = useSelect( ( select ) =>
+		select( 'core' ).getEntityRecordChangesByRecord()
+	);
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
+
+	const [ checkedById, _setCheckedById ] = useState( () => new EquivalentKeyMap() );
+	const setCheckedById = ( id, checked ) =>
+		_setCheckedById( ( prevCheckedById ) => {
+			const nextCheckedById = new EquivalentKeyMap( prevCheckedById );
+			if ( checked ) {
+				nextCheckedById.set( id, true );
+			} else {
+				nextCheckedById.delete( id );
+			}
+			return nextCheckedById;
+		} );
+	const saveCheckedEntities = () => {
+		checkedById.forEach( ( _checked, id ) => {
+			if ( ! ignoredForSave.has( id ) ) {
+				saveEditedEntityRecord(
+					...id.filter( ( s, i ) => i !== id.length - 1 || s !== 'undefined' )
+				);
+			}
+		} );
+		onRequestClose( checkedById );
+	};
+	return (
+		isOpen && (
+			<Modal
+				title={ __( 'What do you want to save?' ) }
+				onRequestClose={ onRequestClose }
```

Comment:
This won't pass an argument to `closeEntitiesSavedStates`, which causes an error:

```
TypeError: savedById.has is not a function
```

https://github.com/WordPress/gutenberg/blob/ff5da59/packages/editor/src/components/post-publish-button/index.js#L68

Steps:

1. Update Site Title block and post title
2. Press "Publish..."
3. Press <kbd>Escape</kbd>

Reply:
Fixed.